### ISSUE TITLE
`too many open chunks` issue in `1.8.4` that was fixed in `1.8.9`

### ISSUE THREAD

**Issue by qingling128** [fluent/fluent-bit#4403]
## Bug Report

Note: This is just a tracking bug. The issue has been fixed in `1.8.9`.

**Describe the bug**

Fluent Bit failed to recover from temporary DNS resolution failures and keeps emitting `too many open chunks` errors.

**To Reproduce**

To repro this issue:

### Step 1 - Create and connect to a Windows VM

```
$ gcloud compute instances create --image-project windows-cloud --image windows-server-2016-dc-v20211012 lingshi-windows-customer-issue
```

### Step 2 - Install Ops Agent

Run PowerShell as administrator, then:
```
(New-Object Net.WebClient).DownloadFile("https://dl.google.com/cloudagents/add-google-cloud-ops-agent-repo.ps1", "${env:UserProfile}\add-google-cloud-ops-agent-repo.ps1")
Invoke-Expression "${env:UserProfile}\add-google-cloud-ops-agent-repo.ps1 -AlsoInstall"
```

### Step 3 - Mess up the DNS

![image](https://user-images.githubusercontent.com/5287526/145118716-5fb3078f-296a-4526-b113-a256443a118d.png)
![image](https://user-images.githubusercontent.com/5287526/145118730-d862cca9-beda-483a-bd7d-7003484acb7e.png)
![image](https://user-images.githubusercontent.com/5287526/145118734-32b5c78b-fe0b-4d51-ac3b-bf4d0b2dcac2.png)
![image](https://user-images.githubusercontent.com/5287526/145118738-2f351781-5dfe-4602-9c27-002d959efba3.png)

Observed problematic log: 

```
[2021/11/03 00:13:27] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:27] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:27] [error] [storage] [cio file] too many open chunks on 'tail.1/1424-1635898405.678490400.flb' (cio_file_up() line=791)
[2021/11/03 00:13:27] [error] [storage] [cio file] too many open chunks on 'emitter.3/1424-1635898406.717911200.flb' (cio_file_up() line=791)
[2021/11/03 00:13:31] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:32] [error] [storage] [cio file] too many open chunks on 'tail.1/1424-1635898405.678490400.flb' (cio_file_up() line=791)
[2021/11/03 00:13:32] [error] [storage] [cio file] too many open chunks on 'emitter.3/1424-1635898406.717911200.flb' (cio_file_up() line=791)
[2021/11/03 00:13:33] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:33] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:33] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:33] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:33] [error] [storage] [cio file] too many open chunks on 'tail.1/1424-1635898405.678490400.flb' (cio_file_up() line=791)
[2021/11/03 00:13:33] [error] [storage] [cio file] too many open chunks on 'emitter.3/1424-1635898406.717911200.flb' (cio_file_up() line=791)
[2021/11/03 00:13:34] [error] [storage] [cio file] too many open chunks on 'tail.1/1424-1635898405.678490400.flb' (cio_file_up() line=791)
[2021/11/03 00:13:34] [error] [storage] [cio file] too many open chunks on 'emitter.3/1424-1635898406.717911200.flb' (cio_file_up() line=791)
[2021/11/03 00:13:34] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
[2021/11/03 00:13:34] [error] [upstream] connection #-1 to logging.googleapis.com:443 timed out after 10 seconds
```

### Step 5 - Restore the DNS setting

Observed the problematic logs never stopped.

Full logs: 
[debug logs.txt](https://github.com/fluent/fluent-bit/files/7672131/debug.logs.txt)

**Comment by jkschulz** [comment]
I think this was fixed in 1.8.8: 
![image](https://user-images.githubusercontent.com/35477443/145938573-54082652-24d1-4387-8ad2-b1fe7c9af485.png)

**Comment by jkschulz** [comment]
Not clear which of these actually fixed it (http://screen/726zLnitrVhUETb): 
![image](https://user-images.githubusercontent.com/35477443/145939691-c4ace25d-01a2-4841-9add-e1ae8c2c8a77.png)
